src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeValueProvider.cs

```diff
@@ -137,6 +137,12 @@ public IValueProvider Filter(BindingSource bindingSource)
                 }
             }
 
+            if (filteredValueProviders.Count == Count)
```

Comment:
It if makes it easier for testing then fine. This is exactly the kind of thing we don't do for perf reasons without evidence though.